pkg/controller/garbagecollector/patch.go

```diff
@@ -52,3 +52,8 @@ func (n *node) patchToUnblockOwnerReferences() ([]byte, error) {
 	dummy.ObjectMeta.UID = n.identity.UID
 	return json.Marshal(dummy)
 }
+
+func deleteFinalizerPatch(finalizer string, UID types.UID) []byte {
+	patch := fmt.Sprintf(`{"metadata":{"$deleteFromPrimitiveList/finalizers":["%s"],"uid":"%s"}}`, finalizer, UID)
```

Comment:
this assumed strategic merge patch is supported on the resource, which is not a valid assumption (see https://github.com/kubernetes/kubernetes/issues/56348)

Reply:
Noted.